scripts/gnosis/scrape_tokens.py

```diff
@@ -15,46 +15,47 @@ def main():
     """
     generate a csv airdrop of all tokens and their balances from `treasurey_tokens` for a given address
     csv saved to scripts/gnosis/<address>_token_balances.csv
-    
+
     `scrape` can be called directly from command line with an address:
     brownie run scrape_tokens.py scrape <address> <receiver>
     """
-    
+
     wallet_address = ''
     receiver=''
     scrape(wallet_address, receiver)
 
 def scrape(address, receiver=''):
     with Progress() as progress:
-        token_data = {'token_type': [], 'token_address': [], 'receiver': [], 'value': [], 'id': [], 'decimals': []}
-        
-        scraping = progress.add_task("[yellow]Scraping...", total=len(registry.eth.treasury_tokens))
+        token_list = list(registry.eth.treasury_tokens.values()) + \
+            list(registry.eth.sett_vaults.values()) + \
+            list(registry.eth.rari.values())
+        token_data = {'token_type': [], 'token_address': [], 'receiver': [], 'value': [], 'id': []}
+
+        scraping = progress.add_task("[yellow]Scraping...", total=len(token_list))
         table = Table()
-        
+
         for key in token_data:
             table.add_column(key, justify="right", style="bright_yellow")
-            
-        for token_name in registry.eth.treasury_tokens:
-            token = interface.ERC20(registry.eth.treasury_tokens[token_name])
-            decimals = token.decimals()
+
+        for token_addr in token_list:
+            if token_addr in [registry.eth.rari.dai_manager, registry.eth.rari.unitroller]:
+                continue
+            token = Contract(token_addr)
```

Comment:
running this for the first time takes about ~5 minutes. Any reason to use `Contract` over `interface.ERC20` here?

Reply:
the problem with interface.erc20 in this case is that there are also lp tokens included in that for loop. this means that if you call the lp token at a later time again with Contract(), it will revert to that interface.erc20 it was locally saved as, and miss attributes.